runtime-transaction/src/program_id_flags.rs

```diff
@@ -0,0 +1,75 @@
+use {
+    solana_pubkey::Pubkey,
+    solana_sdk::{compute_budget, ed25519_program, secp256k1_program},
+};
+
+/// Top-level program id flags.
+/// The main checking point is done with this enum, and for individual
+/// processors, the flags are converted to specific sub-enums.
+#[derive(Clone)]
+pub enum ProgramIdFlag {
+    NoMatch,
+    Secp256k1,
+    Ed25519,
+    ComputeBudgetProgram,
+}
+
+/// Translate from a [Pubkey] to a [ProgramIdFlag].
+impl From<&Pubkey> for ProgramIdFlag {
+    #[inline]
+    fn from(program_id: &Pubkey) -> Self {
+        if program_id == &secp256k1_program::ID {
+            Self::Secp256k1
+        } else if program_id == &ed25519_program::ID {
+            Self::Ed25519
+        } else if program_id == &compute_budget::ID {
+            Self::ComputeBudgetProgram
+        } else {
+            Self::NoMatch
+        }
+    }
+}
+
+/// Self translation for [ProgramIdFlag].
+impl From<&ProgramIdFlag> for ProgramIdFlag {
```

Comment:
why need this?

Reply:
conforming with my generic code. May be able to do some sort of better generic in the code, but copying a u8 for now seems fine to me.